### PR TITLE
Refactoring flux fits

### LINKED ISSUE #663: Fit of resonator flux hangs
I am not sure if that is platform dependent or a general problem but when running on qw5q_gold (using qibolab, qibocal and platforms_qrc main) the resonator flux routine hangs, I believe, on the fitting part (but could also be the plotting). 

If I cancel (Ctrl+C) the job and then do `qq report` on the acquired data I get the report properly.

### PR DESCRIPTION
**Pull request by andrea-pasquale**
This PR closes #663 #448.

I'm trying to improve the performance of the fits by doing the following:

1. Replacing `image_to_curve` with a SNR filter
2. Simplify the fitting functions
3. Putting tight bounds for the fits

Here are the some reports with the new fits:
http://login.qrccluster.com:9000/hjpBOk4dRIWPeWoXADLWNQ==/
http://login.qrccluster.com:9000/Xp4RROS0QCKoDR5nhyBxQw==/

TODOs:

- [x] Fix tables and update methods
- [x] ~Re-introduce 2 order fit (do we really want it?)~
- [x] Fix qubit tracking
- [x] Compute matrix element $V_{ii}$

Checklist:
- [x] Reviewers confirm new code works as expected.
- [x] Tests are passing.
- [x] Coverage does not decrease.
- [x] Documentation is updated.
- [x] Compatibility with Qibo modules (Please edit this section if the current pull request is not compatible with the following branches).
    - [x] Qibo: `master`
    - [x] Qibolab: `main`
    - [x] Qibolab_platforms_qrc: `main`

### CODE REVIEW

File: src/qibocal/protocols/characterization/flux_dependence/utils.py [hunk 1]
```diff
@@ -275,280 +170,304 @@ def flux_crosstalk_plot(data, qubit):
...
-            bias,
-            brf,
-            g,
-            sweetspot,
-            flux_to_bias,
-            asymmetry,
-            Ec,
-            Ej,
-        )
+# def mathieu(index, x):
```
Comment (Jacfomg) [inline src/qibocal/protocols/characterization/flux_dependence/utils.py:288]: Can we remove all this commented code ?

File: src/qibocal/protocols/characterization/flux_dependence/utils.py [hunk 2]
```diff
@@ -275,280 +170,304 @@ def flux_crosstalk_plot(data, qubit):
...
+#     Returns:
+#         (float)
+#     """
+#     return p3 * G_f_d(x, p0, p1, p2)
 
 
-def feature(x, order=3):
-    """
-    Auxilliary function for the function image_to_curve(). It generates a polynomial feature of the form [1, x, x^2, ..., x^order].
+# def freq_r_transmon(x, p0, p1, p2, p3, p4, p5):
```
Comment (Jacfomg) [inline src/qibocal/protocols/characterization/flux_dependence/utils.py:252]: And this one ?

File: src/qibocal/protocols/characterization/flux_dependence/utils.py [hunk 3]
```diff
@@ -275,280 +170,304 @@ def flux_crosstalk_plot(data, qubit):
...
-    return f_r
+    return resonator_freq + g**2 * G_f_d(x, offset, d, element) / (
+        resonator_freq - transmon_frequency(x, w_max, d, element, offset)
+    )
 
 
-def line(x, p0, p1):
-    """
-    Linear fit.
+# TODO: restore second order approximation
```
Comment (Jacfomg) [inline src/qibocal/protocols/characterization/flux_dependence/utils.py:235]: Can we test if we need that approximation ?

File: src/qibocal/protocols/characterization/flux_dependence/resonator_flux_dependence.py [hunk 4]
```diff
@@ -184,226 +164,120 @@ def _fit(data: ResonatorFluxData) -> ResonatorFluxResults:
...
-    brf = {}
-    ssf_brf = {}
-    ECs = {}
-    EJs = {}
-
+    d = {}
+    bare_frequency = {}
+    drive_frequency = {}
     fitted_parameters = {}
+    matrix_element = {}
```
Comment (Jacfomg) [inline src/qibocal/protocols/characterization/flux_dependence/resonator_flux_dependence.py:171]: It is slighty confusing that you use both cii and matrix_element to refer to the same thing if I understood correctly.

File: src/qibocal/protocols/characterization/flux_dependence/resonator_flux_dependence.py [hunk 5]
```diff
@@ -184,226 +164,120 @@ def _fit(data: ResonatorFluxData) -> ResonatorFluxResults:
...
+                    np.mean(qubit_data.freq) / 1e9 - 1,
+                    0,
+                ],
+                [
+                    data.qubit_frequency[qubit] / 1e9 + 0.5,
+                    1,
+                    np.inf,
+                    np.mean(qubit_data.bias) + 0.5,
+                    np.mean(qubit_data.freq) / 1e9 + 1,
+                    1,
```
Comment (Jacfomg) [inline src/qibocal/protocols/characterization/flux_dependence/resonator_flux_dependence.py:213]: we could define some common bounds changing the last element ?

File: src/qibocal/protocols/characterization/flux_dependence/resonator_flux_dependence.py (hunk 4 above)
Comment (andrea-pasquale) [reply src/qibocal/protocols/characterization/flux_dependence/resonator_flux_dependence.py:171]: You are right.

**Review by AlejandroSopena — APPROVED** [review]
Thanks @andrea-pasquale. The first order fit seems good enough.

**Review by Jacfomg — APPROVED** [review]

**Pull request merged**